src/python/pants/backend/experimental/python/lockfile_metadata.py

```diff
@@ -92,75 +109,11 @@ def is_valid_for(
         )
 
 
-def calculate_invalidation_digest(
-    requirements: FrozenOrderedSet[str],
-) -> str:
+def calculate_invalidation_digest(requirements: Iterable[str]) -> str:
```

Comment:
`Iterable` is too general a type here, because the digest function requires a predictable iteration order; `Iterable` doesn't guarantee iteration order. Is there a more general type than `FrozenOrderedSet` that suits the requirements here? 

Alternatively, we can change the digest function to sort the requirements first and regenerate all the lockfiles and associated tests.

Reply:
> we can change the digest function to sort the requirements first

That's a good suggestion. Right now, none of the call sites are sorted. (Although they are stable). I think sorting will make the metadata we would add in https://github.com/pantsbuild/pants/issues/12610 more presentable. 

I'll make this change in a followup because it will require regenerating all the lockfiles and I want to keep this PR more focused. Going to keep this change, given the fast follow - nothing is broken code-wise, only the type hint is not as accurate as need be.